### ISSUE TITLE
Typescript 2.9 compatibility

### ISSUE THREAD

**Issue by jethrolarson** [types/npm-ramda#391]
Looks like TS 2.9 changed the way `keyof`works and now a bunch of errors like the below are showing up:

```
node_modules/@types/ramda/src/mergeWithKey.d.ts:78:61 - error TS2344: Type 'keyof T' does not satisfy the constraint 'string'.
  Type 'string | number | symbol' is not assignable to type 'string'.
    Type 'number' is not assignable to type 'string'.
```

**Comment by jethrolarson** [comment]
https://blogs.msdn.microsoft.com/typescript/2018/05/31/announcing-typescript-2-9/#breaking-changes has the deets.

**Comment by tbrisbane** [comment]
Ran into this over the weekend; it's stalling our ability to upgrade to TS 2.9 in a few of our microservices 🙁

**Comment by strax** [comment]
It's possible to use the previous behavior of `keyof` and get rid of these warnings until typings are updated by adding the following configuration to `tsconfig.json`:

```
{
    "compilerOptions": {
        "keyofStringsOnly": true
    }
}
```

**Comment by ikatyang** [comment]
It should be possible, unless you used some tuple types. The [fixed length tuple](https://blogs.msdn.microsoft.com/typescript/2018/01/17/announcing-typescript-2-7-rc/#fixed-length-tuples) is another incompatible part and it couldn't be disabled. I've opened #399 to address these issues but I'm not sure if it addressed all the TS 2.9 issues, at least there's no warning in generated definitions now and tests are passed.

**Comment by tycho01** [comment]
> The fixed length tuple is another incompatible part and it couldn't be disabled.

That one is my fault. You're welcome. :D
